text/0000-fn_name_macro.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: fn_name_macro
+- Start Date: 2019-11-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+<!-- https://github.com/rust-lang/rfcs/issues/1743 -->
+
+# Summary
+[summary]: #summary
+
+This RFC adds an additional macro, `function!`, to the `core` crate. When invoked, the macro expands to the name of the function that contains the call site.
+
+# Motivation
+[motivation]: #motivation
+
+This is a useful extension of Rust's existing debug reporting: `file!`, `line!`, `column!` and `module_path!`.
+
+For most people the name of the function is a much more immediate way to understand the context of a message than file name and line number.
```

Comment:
Also I'm not sure I agree with this, especially not when the claim is "much more". In particular, the `dbg!()` output is structured in such a way that e.g. in VSCode's integrated terminal, the output becomes a clickable link that immediately takes you to the source location. That seems hard to beat...?

Reply:
That assumes that you are looking at the message in the VS Code integrated terminal. One of my machines doesn't even run VS Code.

Often I see panic or log messages that are pasted into Discord/Zulip/Riot/Github, where you also cannot click the `file:line` text.